src/modules/user/user.controller.ts

```diff
@@ -150,4 +150,24 @@ export class UserController {
   refreshToken(@UserDecorator() user: User) {
     return this.userUseCases.getAuthTokens(user);
   }
+
+  @Post('/search')
+  @HttpCode(200)
+  @ApiOperation({
+    summary: 'Search items by name',
+  })
+  async searchItemsByName(
+    @UserDecorator() user: User,
+    @Body() body: { plain_name: string },
+  ) {
+    if (!user || !body || !body.plain_name) {
+      throw new NotFoundException();
```

Comment:
It is a good idea to request HTTP Status Code Semantics, use `BadRequestException`